### PR TITLE
docs(notifications) add approval_requested reason type

### LINKED ISSUE #31691: docs: Notifications REST API missing `approval_requested` reason
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#about-notification-reasons

### What changes are you suggesting?

Add the reason `approval_requested`, which is went for environment deployment approvals

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by setchy**
### Why:

Closes: https://github.com/github/docs/issues/31691

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Document the `approval_requested` notification reason type.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@setchy Thanks for opening a PR! ✨  

Please fill out the [pull request template](https://raw.githubusercontent.com/github/docs/main/.github/PULL_REQUEST_TEMPLATE.md) in the top comment in this pull request, then we'll get this up for review!

**Comment by setchy** [comment]
Sure thing.  All updated @nguyenalex836

**Comment by nguyenalex836** [comment]
@setchy Beautiful, thank you for doing that! ✨ I'll get this up for review!

File: content/rest/activity/notifications.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ There are a few potential `reason`s for receiving a notification.
 
 Reason Name | Description
 ------------|------------
+`approval_requested` | You were requested to review and approve a deployment.
```
Comment (skedwards88) [inline content/rest/activity/notifications.md:51]: Let's also add a link to more info:
```suggestion
`approval_requested` | You were requested to review and approve a deployment. For more information, see "[AUTOTITLE](/actions/managing-workflow-runs/reviewing-deployments)."
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for this addition! I made one suggestion that I'll incorporate, then we'll get this merged for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**